crates/bevy_tasks/src/task_pool.rs

```diff
@@ -230,23 +311,46 @@ impl TaskPool {
                         break result;
                     };
 
-                    self.executor.try_tick();
+                    executor.try_tick();
                     local_executor.try_tick();
                 }
             }
         })
     }
 
-    /// Spawns a static future onto the thread pool. The returned Task is a future. It can also be
-    /// cancelled and "detached" allowing it to continue running without having to be polled by the
-    /// end-user.
+    /// Spawns a static future onto the thread pool in the [`Compute`] group. The returned Task is a future.
+    /// It can also be cancelled and "detached" allowing it to continue running without having to be polled
+    /// by the end-user.
     ///
     /// If the provided future is non-`Send`, [`TaskPool::spawn_local`] should be used instead.
+    ///
+    /// [`Compute`]: crate::TaskGroup::Compute
     pub fn spawn<T>(&self, future: impl Future<Output = T> + Send + 'static) -> Task<T>
     where
         T: Send + 'static,
     {
-        Task::new(self.executor.spawn(future))
+        self.spawn_as(TaskGroup::Compute, future)
```

Comment:
This was mostly chosen to avoid breakages elsewhere. The primary use of the task pool internally is with the `ComputeTaskPool` and `par_for_each`. The other two are straightforwardly broken by this change already by removing them, but the use of `TaskPool::spawn` in the typical case is for compute right now. We could alternatively remove this entirely and force people to use `spawn_as` and explicitly state which group they prefer.

Reply:
I'd be down to remove it. It wouldn't be too different from the state today where users have to pick which task pool to spawn on. I think most users when they reach for the task pools are doing so to spawn longer running tasks and they'll reach for `spawn` before `spawn_as`, which would be a mistake.